### ISSUE TITLE
Please consider removing external libs

### ISSUE THREAD

**Issue by pcav** [QGIS-Contribution/QGIS-ResourceSharing#112]
They can be included in installers for some OSs (e.g. OSGeo4W)

**Comment by havatv** [comment]
@pcav, thank you for showing interest in the plugin, and for providing input!
If external libs that the plugin depends on can not be guaranteed to be included for all platforms and all situations, removing them from the plugin means that the plugin would not work.

**Comment by pcav** [comment]
Yes, of course. The usual course for other plugins is:
* to include the libs in osgeo4w and provide instructions to install on Linux (usually very esy)
* if this is not possible (usually because they are very specialized, of very limited use), just provide instructions to install the libs before installing the plugin.

**Comment by havatv** [comment]
> Yes, of course. The usual course for other plugins is:
> 
>     * to include the libs in osgeo4w and provide instructions to install on Linux (usually very esy)
> 
>     * if this is not possible (usually because they are very specialized, of very limited use), just provide instructions to install the libs before installing the plugin.

This sounds easy, but my experience (in a university setting) is that users are not allowed to install anything on the university lab computer or the university personal computers, and most of the 300 students on our introductory GIS course would fail to install any Python library (they have no interest in - or knowledge about the technicalities of the software on their computers). And I think we should avoid setting up barriers for QGIS Resource Sharing.

**Comment by pcav** [comment]
Agreed -that's why the first choice is to add them to OSGeo4W

**Comment by havatv** [comment]
> Agreed -that's why the first choice is to add them to OSGeo4W

That does not help for the Mac users, and there are quite a number of them among the students.

**Comment by pcav** [comment]
adding it to the mac installer as well is quite feasible, and it will help the plugin inclusion in core (which is something I believe should happen)

**Comment by havatv** [comment]
I very much agree that we should avoid including external libraries if we can, and I am ready to remove them as soon as these libraries are available with a default QGIS install on all platforms. My "sin" here was to add the *pathvalidate* package. The other packages (dulwich for github access and giturlparse for parsing git URLs) were added long before I "took over" the development of the plugin.

**Comment by pcav** [comment]
No sin! :)
Could you please open a ticket on osgeo4w and osx packaging asking to include these libs?
Thanks

**Comment by Guts** [comment]
Submitting the required libs to the packagers would not be enough. The libraries would still have to be included into **default** packages to keep the plugin usable by the majority  and I don't think that's imaginable.

On the one hand because they are only interesting for specific functions, especially related to the functional scope of this plugin. On the other hand, the QGIS MSI  installer now weighs about 1 GB due to the inclusion of PROJ 8. It's time to go on a diet!

There are many discussions, in [QEPs](https://github.com/qgis/QGIS-Enhancement-Proposals/issues?q=is%3Aissue+pip+) or developer forums, about the possibility to integrate python dependencies management. And for the moment, there are as many workarounds (wild pip installs in subcall) as refusals from QGIS devs!

In short, integrating the dependencies in the plugin packaging seems to be the best solution for the moment.
